gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -552,16 +564,52 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
                 self._use_mtls_endpoint))
 
         if not transport_provided:
+            {# TODO(https://github.com/googleapis/gapic-generator-python/issues/2121): Remove this condition when async rest is GA. #}
+            {% if rest_async_io_enabled %}
+            transport_init: Union[Type[{{ service.name  }}Transport], Callable[..., {{ service.name }}Transport]] = (
+                {{ service.client_name  }}.get_transport_class(transport)
+                if isinstance(transport, str) or transport is None
+                else cast(Callable[..., {{ service.name }}Transport], transport)
+            )
+
+            if "rest_asyncio" in str(transport_init):
+                # Note: The following parameters are not support in async rest:
+                # credentials_file, scopes, client_cert_source_for_mtls, quota_project_id
+                # always_use_jwt_access, api_audience.
+                unsupported_params = {
+                    "google.api_core.client_options.ClientOptions.credentials_file": self._client_options.credentials_file,
+                    "google.api_core.client_options.ClientOptions.scopes": self._client_options.scopes,
+                    "google.api_core.client_options.ClientOptions.quota_project_id": self._client_options.quota_project_id,
+                    "google.api_core.client_options.ClientOptions.client_cert_source": self._client_options.client_cert_source,
+                    "google.api_core.client_options.ClientOptions.api_audience": self._client_options.api_audience,
+
+                }
+                provided_unsupported_params = [name for name, value in unsupported_params.items() if value is not None]
+                if provided_unsupported_params:
+                    raise NotImplementedError(
+                        f"The following provided parameters are not supported for `transport=rest_asyncio`: {', '.join(provided_unsupported_params)}"
+                    )
+                self._transport = transport_init(
+                    credentials=credentials,
+                    host=self._api_endpoint,
+                    client_info=client_info,
+                )
+                return
+
+            {% endif %}
             import google.auth._default  # type: ignore
 
             if api_key_value and hasattr(google.auth._default, "get_api_key_credentials"):
                 credentials = google.auth._default.get_api_key_credentials(api_key_value)
 
+            {# TODO(https://github.com/googleapis/gapic-generator-python/issues/2121): Remove this condition when async rest is GA. #}
+            {% if not rest_async_io_enabled %}
```

Comment:
Why the `not` here? The init was working for non-sync-rest before, right?You'll have a different constructor for AsyncRest? If so, I would suggest tweaking to make this explicit. For example, something like:

```suggestion
           {% if rest_async_io_enabled %}
              raise NotImplementedError("transport_init for async REST not implemented yet")
           {% else %}
            transport_init: ...
            {% endif %}
```

(Or clarify if I interpreted this wrong.)

Reply:
`transport_init` is already defined  here`https://github.com/googleapis/gapic-generator-python/pull/2130/files/a2d897a53d09facf9bbf8e2ae49d683ef5963cc4#diff-c71dde17175c66932820c2ad6ba1347a2df51eccb8d39b9060abb007bb0e5f63R569` when async rest is enabled so we don't need to re-initialize it here.

I could just generate `transport_init` unconditionally like it was before but I needed it above to do some checks. And not doing it the way i'm doing would regenerate every client. I was gonna do it as a follow-up so that I could keep the diffs for the generated clients smaller.

Let me know if this makes sense.

Reply:
Ah, OK. Makes sense.